lib/pf/services/manager/firewalld.pm

```diff
@@ -0,0 +1,187 @@
+package pf::services::manager::firewalld;
+
+=head1 NAME
+
+pf::services::manager::firewalld add documentation
+
+=cut
+
+=head1 DESCRIPTION
+
+pf::services::manager::firewalld
+
+=cut
+
+use strict;
+use warnings;
+use Moo;
+use pf::file_paths qw($install_dir);
+use pf::log;
+use pf::util;
+use pf::firewalld;
+use pf::config qw(%Config);
+
+extends 'pf::services::manager';
+
+has '+name' => (default => sub { 'firewalld' } );
+
+has '+shouldCheckup' => ( default => sub { 1 }  );
+
+has 'runningServices' => (is => 'rw', default => sub { 0 } );
+
+
+=head2 start
+
+start firewalld
+
+=cut
+
+sub startService {
+    my ($self) = @_;
+    my $technique;
+    unless ($self->isAlive()) {
+        #$technique = getIptablesTechnique();
+        #$technique->iptables_save($install_dir . '/var/iptables.bak');
+    }
+    #$technique ||= getIptablesTechnique();
+    #$technique->iptables_generate();
+    return 1;
+}
+
+=head2
+
+generateConfig
+
+=cut
+
+sub generateConfig {
+    my $technique;
+    #$technique ||= getIptablesTechnique();
```

Comment:
i think here you should call pf::firewalld::firewalld_generate_configs() and in the systemd script you should launch this function

Reply:
Hum, Ok now I get it. I directly call it from the systemd.
It needs to be done in two times. One pre and one post.
I will check how to use it.
Thanks